File: server/PQL-Domain-Submission/src/main/java/com/passql/submission/dto/ReadinessResponse.java
package com.passql.submission.dto;

import com.passql.submission.readiness.ToneKey;

import java.time.LocalDateTime;

/**
 * 합격 준비도(Readiness) 응답 블록 (v3).
 *
 * base(Accuracy × Coverage × Recency) + bonus(retry, spread) 구조의 5요소 원본값 공개.
 */
public record ReadinessResponse(
    double score,
    double accuracy,
    double coverage,
    double recency,
    double retry,
    double spread,
    LocalDateTime lastStudiedAt,
    int recentAttemptCount,
    int coveredTopicCount,
    int activeTopicCount,
    Integer daysUntilExam,
    ToneKey toneKey
) {}

File: app/lib/data/models/progress/readiness_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'readiness_response.freezed.dart';
part 'readiness_response.g.dart';

/// 합격 준비도 블록.
/// score = accuracy × coverage × recency (0.0~1.0)
/// toneKey: NO_EXAM | ONBOARDING | POST_EXAM | TODAY | SPRINT | PUSH | STEADY | EARLY
@freezed
class ReadinessResponse with _$ReadinessResponse {
  const factory ReadinessResponse({
    required double score,
    required double accuracy,
    required double coverage,
    required double recency,
    String? lastStudiedAt,
    required int recentAttemptCount,
    required int coveredTopicCount,
    required int activeTopicCount,
    int? daysUntilExam,
    required String toneKey,
  }) = _ReadinessResponse;

  factory ReadinessResponse.fromJson(Map<String, dynamic> json) =>
      _$ReadinessResponseFromJson(json);
}
